packages/amplify-provider-awscloudformation/src/aws-utils/aws-iam.ts

```diff
@@ -0,0 +1,20 @@
+import aws from './aws.js';
+import awstype from 'aws-sdk';
+import { IAM } from 'aws-sdk';
+import { AwsSdkConfig } from '../utils/auth-types.js';
+
+let instance: IAM;
+
+export const getIAMClient = async (sdkConfigProvider: () => Promise<AwsSdkConfig>) => {
```

Comment:
Could we change this to already existing pattern of getting instance of service client. I think there are already 3 patterns and adding this would make it the 4th pattern.